### ISSUE TITLE
indent-binary-ops spaces config not working

### ISSUE THREAD

**Issue by rgant** [eslint-stylistic/eslint-stylistic#226]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The spaces configuration for `indent-binary-ops` doesn't appear to have any effect on the indentation expected.

With config: `"@stylistic/indent-binary-ops": ["error", 2],` I get this error:

```
.../src/app/auth/login/email-login/email-login.component.spec.ts
  468:1  error  Expected indentation of 6 spaces  @stylistic/indent-binary-ops
```

With config: `"@stylistic/indent-binary-ops": ["error", 22],` I get the same error:

```
.../src/app/auth/login/email-login/email-login.component.spec.ts
  468:1  error  Expected indentation of 6 spaces  @stylistic/indent-binary-ops
```

I would love to submit a PR with broken tests, but I don't see how to set the rule config in the tests for `indent-binary-ops`.

I would love to submit a StackBlitz reproduction, but after waiting 20 minutes for "Installing dependencies" to finish I am going to bed soon.

I guess I'll try again later.

Possibly related: #222

### Reproduction

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-xhfb1w?file=playground%2Feslint.config.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests

**Comment by antfu** [comment]
`indent-binary-ops` works by aligning with the indentation of the previous line and increasing the level under certain conditions. In your cases it does not match the rule of level-increasing.

**Comment by rgant** [comment]
I don't understand how I can say "indentation should be multiples of 22" and get back an error that says that I need 6 spaces of indentation.

**Comment by antfu** [comment]
You will need to have it with `@stylistic/indent: ['error', 22]` then you will have what you expected

**Comment by rgant** [comment]
The error is not with @stylistic/indent, but the indentation of "binary" operators.

```ts
function foobar() {
  const somestring = 'foo'
    + 'bar';
}
```

Is the indentation I want. I have `@stylistic/indent: ['error', 2]`. I have `"@stylistic/indent-binary-ops": ["error", 2]`. The above code does not pass lint.

The error I get is `Expected indentation of 6 spaces  @stylistic/indent-binary-ops`. 6 makes no sense to me as the outcome of that setup. It makes even less sense with `@stylistic/indent: ['error', 2], "@stylistic/indent-binary-ops": ["error", 22]`. There is either a flaw in the code or the documentation. 6 spaces does not appear reasonable for either configuration.

**Comment by Eugeno** [comment]
The rule `@stylistic/plus/indent-binary-ops` works correctly in some situations and incorrectly in others:
![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/23382920/345883a7-d880-4871-bdeb-5479ec638a51)
v1.5.3

**Comment by Eugeno** [comment]
New bug in 1.6.0:
![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/23382920/7916cd08-42b3-4a41-bda8-6c3f551fae80)

**Comment by SamanthaAdrichem** [comment]
This issue also exists with "@stylistic/eslint-plugin": "1.7.0", and &&

```
test() {
    this.a = true === this.b
        && false === this.c
}
```

will error and expect
```
test() {
    this.a = true === this.b
    && false === this.c
}
```

my current work-around is
```
test() {
    this.a = 
        true === this.b
        && false === this.c
}
```

But that's just ugly 😓

**Comment by Eugeno** [comment]
v2.2.2: still relevant

Reproduction:
https://stackblitz.com/edit/plusindent-binary-ops-bug-db2s9h?file=src%2Findex.ts

@antfu please note

**Comment by Xenfo** [comment]
I opened #469 for this issue.
